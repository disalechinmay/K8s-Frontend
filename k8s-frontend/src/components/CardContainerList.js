import React, { Component } from "react";
import { ListGroup } from "react-bootstrap";

class CardContainerList extends Component {
	render() {
		return (
			<React.Fragment>
				<ListGroup variant="flush">
					{this.props.list.map((val, index) => {
						return (
							<ListGroup.Item key={index + "_LG_ITEM"}>
								<span className="text-muted">
									Container #{index + 1}:
								</span>{" "}
								{val}
							</ListGroup.Item>
						);
					})}
				</ListGroup>
			</React.Fragment>
		);
	}
}

export default CardContainerList;
