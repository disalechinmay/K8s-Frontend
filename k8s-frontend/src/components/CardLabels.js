import React, { Component } from "react";
import { Button, Badge } from "react-bootstrap";

class CardLabels extends Component {
	render() {
		return (
			<React.Fragment>
				{this.props.labels &&
					Object.entries(this.props.labels).map(
						([key, val], index) => {
							return (
								<React.Fragment key={index + "_LABEL_FRAG"}>
									<Button
										variant={
											(this.props.matchedLabels &&
												this.props.matchedLabels.hasOwnProperty(
													key
												) &&
												val ===
													this.props.matchedLabels[
														key
													] &&
												"success") ||
											"secondary"
										}
										size="sm"
										className="mb-1 text-wrap"
										key={index + "_LABEL_BUTTON"}
									>
										{key}{" "}
										<Badge
											variant="light"
											key={index + "_LABEL_BADGE"}
											className="text-wrap"
										>
											{val}
										</Badge>
									</Button>
									&nbsp;&nbsp;
								</React.Fragment>
							);
						}
					)}
			</React.Fragment>
		);
	}
}

export default CardLabels;
