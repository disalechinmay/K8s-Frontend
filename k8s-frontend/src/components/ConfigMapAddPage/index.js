export { default as ConfigMapAddPage } from "./ConfigMapAddPage";
