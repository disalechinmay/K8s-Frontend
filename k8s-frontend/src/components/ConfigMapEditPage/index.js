export { default as ConfigMapEditPage } from "./ConfigMapEditPage";
