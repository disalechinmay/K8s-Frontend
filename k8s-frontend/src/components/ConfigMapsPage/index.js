export { default as ConfigMapsPage } from "./ConfigMapsPage";
export { default as ConfigMapCard } from "./ConfigMapCard";
