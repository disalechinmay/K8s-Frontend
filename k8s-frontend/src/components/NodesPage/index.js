export { default as NodesPage } from "./NodesPage";
export { default as NodeCard } from "./NodeCard";
