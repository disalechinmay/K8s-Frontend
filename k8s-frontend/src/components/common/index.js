export { default as CardContainerList } from "./CardContainerList";
export { default as CardLabels } from "./CardLabels";
export { default as LoadingPage } from "./LoadingPage";
export { default as SmallLoadingPage } from "./SmallLoadingPage";
export { default as ErrorPage } from "./ErrorPage";
export { default as SmallErrorPage } from "./SmallErrorPage";
