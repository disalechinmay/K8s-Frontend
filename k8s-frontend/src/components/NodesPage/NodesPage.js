import React, { Component } from "react";
import { SmallLoadingPage, SmallErrorPage } from "../common";
import { getNodes } from "../../services";
import NodeCard from "./NodeCard";

class NodesPage extends Component {
	state = {
		pageLoading: true,
		nodesListSet: false,
		nodesList: [],
		errorSet: false,
		errorDescription: ""
	};
	_isMounted = false;

	componentDidMount() {
		this._isMounted = true;

		// Makes a service call to get a list of nodes
		getNodes()
			.then(result => {
				if (this._isMounted)
					this.setState({
						...this.state,
						pageLoading: false,
						nodesListSet: true,
						nodesList: result.payLoad
					});
			})
			.catch(error => {
				if (this._isMounted)
					this.setState({
						...this.state,
						errorSet: true,
						errorDescription: error
					});
			});
	}

	componentWillUnmount() {
		this._isMounted = false;
	}

	render() {
		if (this.state.errorSet) {
			return (
				<SmallErrorPage
					errorDescription={this.state.errorDescription}
				/>
			);
		}

		if (this.state.pageLoading) return <SmallLoadingPage />;

		return (
			<React.Fragment>
				{this.state.nodesListSet &&
					this.state.nodesList.map((nodeInfo, index) => {
						return (
							<NodeCard
								key={index + "_NODE_CARD"}
								nodeInfo={nodeInfo}
								refreshState={() => this.props.refreshState()}
							/>
						);
					})}
			</React.Fragment>
		);
	}
}

export default NodesPage;
