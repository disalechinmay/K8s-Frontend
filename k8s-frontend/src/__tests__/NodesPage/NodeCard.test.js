import { mount } from "enzyme";
import React from "react";
import toJson from "enzyme-to-json";
import { NodeCard } from "../../components/NodesPage";

describe("NodeCard Tests", () => {
	test("Check if NodeCard doesn't throw error when nodeInfo prop is not passed.", () => {
		let wrapper = mount(<NodeCard />);
		expect(wrapper.isEmptyRender()).toBe(true);
	});

	// Sends in broken nodeInfo prop.
	test("Check if NodeCard doesn't throw error when nodeInfo prop is broken.", () => {
		let nodeInfo = {};
		let wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = { nodeName: {} };
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = { nodeName: "NODE1" };
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = { nodeName: "NODE1", nodeCapacity: "String" };
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = { nodeName: "minikube", nodeCapacity: {} };
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = { nodeName: "minikube", nodeCapacity: { cpu: 5 } };
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = {
			nodeName: "minikube",
			nodeCapacity: { cpu: "5", memory: 5 }
		};
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = {
			nodeName: "minikube",
			nodeCapacity: { cpu: "5", memory: "5", pods: 5 }
		};
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = {
			nodeName: "minikube",
			nodeCapacity: { cpu: "5", memory: "5", pods: "5" }
		};
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);

		nodeInfo = {
			nodeName: "minikube",
			nodeCapacity: {
				cpu: "5",
				memory: "5",
				pods: "5",
				"ephemeral-storage": 5
			}
		};
		wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(wrapper.isEmptyRender()).toBe(true);
	});

	// Checks if everything renders if nodeInfo prop is valid.
	test("Check if everything renders if nodeInfo prop is valid", () => {
		let nodeInfo = {
			nodeAnnotations: {
				"kubeadm.alpha.kubernetes.io/cri-socket":
					"/var/run/dockershim.sock",
				"node.alpha.kubernetes.io/ttl": "0",
				"volumes.kubernetes.io/controller-managed-attach-detach": "true"
			},
			nodeCapacity: {
				cpu: "4",
				"ephemeral-storage": "34471692Ki",
				"hugepages-1Gi": "0",
				"hugepages-2Mi": "0",
				memory: "3963180Ki",
				pods: "110"
			},
			nodeLabels: {
				"beta.kubernetes.io/arch": "amd64",
				"beta.kubernetes.io/os": "linux",
				"kubernetes.io/arch": "amd64",
				"kubernetes.io/hostname": "minikube",
				"kubernetes.io/os": "linux",
				"node-role.kubernetes.io/master": ""
			},
			nodeName: "minikube"
		};

		let wrapper = mount(<NodeCard nodeInfo={nodeInfo} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	test("Check if everything renders if nodeInfo prop is valid & showResourceType is set", () => {
		let nodeInfo = {
			nodeAnnotations: {
				"kubeadm.alpha.kubernetes.io/cri-socket":
					"/var/run/dockershim.sock",
				"node.alpha.kubernetes.io/ttl": "0",
				"volumes.kubernetes.io/controller-managed-attach-detach": "true"
			},
			nodeCapacity: {
				cpu: "4",
				"ephemeral-storage": "34471692Ki",
				"hugepages-1Gi": "0",
				"hugepages-2Mi": "0",
				memory: "3963180Ki",
				pods: "110"
			},
			nodeLabels: {
				"beta.kubernetes.io/arch": "amd64",
				"beta.kubernetes.io/os": "linux",
				"kubernetes.io/arch": "amd64",
				"kubernetes.io/hostname": "minikube",
				"kubernetes.io/os": "linux",
				"node-role.kubernetes.io/master": ""
			},
			nodeName: "minikube"
		};

		let wrapper = mount(
			<NodeCard nodeInfo={nodeInfo} showResourceType={true} />
		);
		expect(wrapper.find(".resource-type").exists()).toBeTruthy();
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
