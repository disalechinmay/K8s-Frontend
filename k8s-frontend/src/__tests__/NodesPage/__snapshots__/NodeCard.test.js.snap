// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeCard Tests Check if everything renders if nodeInfo prop is valid & showResourceType is set 1`] = `
<NodeCard
  nodeInfo={
    Object {
      "nodeAnnotations": Object {
        "kubeadm.alpha.kubernetes.io/cri-socket": "/var/run/dockershim.sock",
        "node.alpha.kubernetes.io/ttl": "0",
        "volumes.kubernetes.io/controller-managed-attach-detach": "true",
      },
      "nodeCapacity": Object {
        "cpu": "4",
        "ephemeral-storage": "34471692Ki",
        "hugepages-1Gi": "0",
        "hugepages-2Mi": "0",
        "memory": "3963180Ki",
        "pods": "110",
      },
      "nodeLabels": Object {
        "beta.kubernetes.io/arch": "amd64",
        "beta.kubernetes.io/os": "linux",
        "kubernetes.io/arch": "amd64",
        "kubernetes.io/hostname": "minikube",
        "kubernetes.io/os": "linux",
        "node-role.kubernetes.io/master": "",
      },
      "nodeName": "minikube",
    }
  }
  showResourceType={true}
>
  <div
    className="card flex flex-column"
  >
    <span
      className="title w-100"
    >
      <span
        className="fa fa-lastfm"
      />
       
      minikube
      <sup
        className="resource-type"
      >
        Node
      </sup>
    </span>
    <div
      className="flex flex-row space-around"
    >
      <span>
        <span
          className="big-number"
        >
          4
        </span>
        <br />
        CPU
      </span>
      <span>
        <span
          className="big-number"
        >
          34471692Ki
        </span>
        <br />
        Storage
      </span>
      <span>
        <span
          className="big-number"
        >
          3963180Ki
        </span>
        <br />
        Memory
      </span>
      <span>
        <span
          className="big-number"
        >
          110
        </span>
        <br />
        Pods
      </span>
    </div>
  </div>
</NodeCard>
`;

exports[`NodeCard Tests Check if everything renders if nodeInfo prop is valid 1`] = `
<NodeCard
  nodeInfo={
    Object {
      "nodeAnnotations": Object {
        "kubeadm.alpha.kubernetes.io/cri-socket": "/var/run/dockershim.sock",
        "node.alpha.kubernetes.io/ttl": "0",
        "volumes.kubernetes.io/controller-managed-attach-detach": "true",
      },
      "nodeCapacity": Object {
        "cpu": "4",
        "ephemeral-storage": "34471692Ki",
        "hugepages-1Gi": "0",
        "hugepages-2Mi": "0",
        "memory": "3963180Ki",
        "pods": "110",
      },
      "nodeLabels": Object {
        "beta.kubernetes.io/arch": "amd64",
        "beta.kubernetes.io/os": "linux",
        "kubernetes.io/arch": "amd64",
        "kubernetes.io/hostname": "minikube",
        "kubernetes.io/os": "linux",
        "node-role.kubernetes.io/master": "",
      },
      "nodeName": "minikube",
    }
  }
>
  <div
    className="card flex flex-column"
  >
    <span
      className="title w-100"
    >
      <span
        className="fa fa-lastfm"
      />
       
      minikube
    </span>
    <div
      className="flex flex-row space-around"
    >
      <span>
        <span
          className="big-number"
        >
          4
        </span>
        <br />
        CPU
      </span>
      <span>
        <span
          className="big-number"
        >
          34471692Ki
        </span>
        <br />
        Storage
      </span>
      <span>
        <span
          className="big-number"
        >
          3963180Ki
        </span>
        <br />
        Memory
      </span>
      <span>
        <span
          className="big-number"
        >
          110
        </span>
        <br />
        Pods
      </span>
    </div>
  </div>
</NodeCard>
`;
